### ISSUE TITLE
Debug tiles have broken "name" fields

### ISSUE THREAD

**Issue by danpat** [Project-OSRM/osrm-backend#3250]
It looks like a recent refactor might have broken the `name` property in the debug tiles.  When looking at results, I can see data like:

```json
{"type":"Feature","id":11,"geometry":{"type":"LineString","coordinates":[[-122.409732341766,37.816242626788],[-122.421748638153,37.8147000552758]]},"properties":{"datasource":"lua profile","duration":1.5,"is_small":false,"name":15.7,"speed":20}},
{"type":"Feature","id":12,"geometry":{"type":"LineString","coordinates":[[-122.421748638153,37.8147000552758],[-122.409732341766,37.816242626788]]},"properties":{"datasource":"lua profile","duration":1.5,"is_small":false,"name":15.7,"speed":20}},
{"type":"Feature","id":13,"geometry":{"type":"LineString","coordinates":[[-122.396793365479,37.8177682152617],[-122.341003417969,37.8247346685399]]},"properties":{"datasource":"lua profile","duration":7.2,"is_small":false,"name":"Polk Street","speed":20}},
{"type":"Feature","id":14,"geometry":{"type":"LineString","coordinates":[[-122.341003417969,37.8247346685399],[-122.396793365479,37.8177682152617]]},"properties":{"datasource":"lua profile","duration":7.2,"is_small":false,"name":"Polk Street","speed":20}}
```

Note that the `name` field in some cases is a float, and in other cases is a string.

Also, the raw tile data often seems to contain the name repeated multiple times, which should not happen (we should only store the name once, and all features that need that name should refer to the same table index).

**Comment by TheMarex** [comment]
Would be a good opportunity for addressing https://github.com/Project-OSRM/osrm-backend/issues/2198

Might be caused by 8205c34a which touched that code.

**Comment by danpat** [comment]
Fixed by #3249
